src/components/ConfigureNodeFormModal/Hysteria2Form.tsx

```diff
@@ -0,0 +1,56 @@
+import { Checkbox, NumberInput, Select, TextInput } from '@mantine/core'
+import { useForm, zodResolver } from '@mantine/form'
+import { useTranslation } from 'react-i18next'
+import { z } from 'zod'
+
+import { FormActions } from '~/components/FormActions'
+import { DEFAULT_HYSTERIA2_FORM_VALUES, hysteria2Schema } from '~/constants'
+import { generateHysteria2URL } from '~/utils'
+
+export const Hysteria2Form = ({ onLinkGeneration }: { onLinkGeneration: (link: string) => void }) => {
+  const { t } = useTranslation()
+  const { onSubmit, getInputProps, reset } = useForm<z.infer<typeof hysteria2Schema>>({
+    initialValues: DEFAULT_HYSTERIA2_FORM_VALUES,
+    validate: zodResolver(hysteria2Schema),
+  })
+
+  const handleSubmit = onSubmit((values) => {
+    /* hysteria2://[auth@]hostname[:port]/?[key=value]&[key=value]... */
+    const query = {
+      obfs: values.obfs,
+      obfsPassword: values.obfsPassword,
+      sni: values.sni,
+      insecure: values.allowInsecure,
+      pinSHA256: values.pinSHA256,
+    }
+
+    return onLinkGeneration(
```

Comment:
Given the dependency chain `daed -> wing -> dae` by git submodule, we must wait for updates before the backend can function. Perhaps we should mark this PR as a draft, considering that CI builds lack actual backend implementation?